src/frontend/src/features/layouts/components/thread-view/components/thread-summary/index.tsx

```diff
@@ -0,0 +1,157 @@
+  threadId,
+  summary,
+  selectedMailboxId,
+  searchParams,
+  selectedThread,
+  onSummaryUpdated,
+}: ThreadSummaryProps) => {
+  const { t } = useTranslation();
+  const [localSummary, setLocalSummary] = useState(summary);
+
+  // Build the cache key for the thread 
+  const threadQueryKey = useMemo(() => {
+    if (!selectedMailboxId || !searchParams) return ["threads"];
+    const queryKey = ["threads", selectedMailboxId];
+    if (searchParams.get("search")) {
+      return [...queryKey, "search"];
+    }
+    return [...queryKey, searchParams.toString()];
+  }, [selectedMailboxId, searchParams]);
+
+  const queryClient = useQueryClient();
+  /**
+   * Cache the new summary in the thread query data.
+   * This is used to update the thread summary in the thread list
+   * when the summary is updated.
+   */
+  const cacheNewSummary = (newSummary: string) => {
+    queryClient.setQueryData(
+      threadQueryKey,
+      (
+        oldData:
+          | {
+              pages: Array<{
+                data: {
+                  results: { id: string; summary?: string }[];
+                  count: number;
+                  next: string | null;
+                  previous: string | null;
+                };
+              }>;
+            }
+          | undefined
+      ) => {
+        if (!oldData) return oldData;
+        return {
+          ...oldData,
+          pages: oldData.pages.map((page) => ({
+            ...page,
+            data: {
+              ...page.data,
+              results: page.data.results.map((thread) =>
+                thread.id === selectedThread?.id
+                  ? { ...thread, summary: newSummary }
+                  : thread
+              ),
+            },
+          })),
+        };
+      }
+    );
+  };
+
+  const refreshMutation = useThreadsRefreshSummaryCreate({
+    mutation: {
+      onMutate: () => {
+        addToast(
+          <ToasterItem type="info">
+            {t("summary.generation_in_progress")}
+          </ToasterItem>
+        );
+      },
+      onSuccess: (data) => {
+        const newSummary =
+          data &&
+          typeof data === "object" &&
+          "data" in data &&
+          data.data &&
+          typeof data.data === "object" &&
+          "summary" in data.data
+            ? (data.data as { summary?: string }).summary
+            : undefined;
+
+        if (newSummary) {
+          setLocalSummary(newSummary);
+          cacheNewSummary(newSummary);
+          onSummaryUpdated?.(newSummary);
+          addToast(<ToasterItem type="info">{t("summary.refresh_success")}</ToasterItem>);
+        }
+      },
+      onError: () => {
+        addToast(
+          <ToasterItem type="error">
+            {t("summary.refresh_error")}
+          </ToasterItem>
+        );
+      },
+    },
+  });
+
+  const handleRefresh = () => {
+    refreshMutation.mutate({ id: threadId });
+  };
+
+  useEffect(() => {
+    setLocalSummary(summary);
+  }, [summary]);
```

Comment:
Is it really useful ?

At mount, the `useState` is hydrated with the `summary` prop value. Then this value should be always up to date when the mutation is called nope ?